src/main/kotlin/org/rust/ide/utils/import/ImportCandidatesCollector2.kt

```diff
@@ -405,6 +399,18 @@ private fun resolveRootPath(defMap: CrateDefMap, path: ItemUsePath, segments: Li
     return perNs
 }
 
+private fun List<RsQualifiedNamedElement>.filterByNamespace(context: ImportContext2): List<RsQualifiedNamedElement> {
+    val pathInfo = context.pathInfo ?: return this
+    return filter {
+        pathInfo.namespaceFilter(it) && checkProcMacroType(it, pathInfo.onlyAttrOrDerive)
+    }
+}
+
+fun checkProcMacroType(element: RsQualifiedNamedElement, onlyAttrOrDerive: Boolean): Boolean {
+    if (onlyAttrOrDerive) return true  /** attr and derives are checked in [RsPath.namespaceFilter] */
+    return if (element is RsFunction && element.isProcMacroDef) element.isBangProcMacroDef else true
+}
```

Comment:
It seems like `onlyAttrOrDerive` is misleading here. That is, this function does a reverse thing - if `onlyAttrOrDerive` is false, it filters function-like macros. So maybe instead pass `onlyFunctionLikeMacros`?

Reply:
I agree. What do you think about renaming it to `parentIsMetaItem` since it is the actual value of the variable?

Reply:
Yep, let's rename it this way